### ISSUE TITLE
Haddock comments needs to be separated by semi-colons when explicit layout is used

### ISSUE THREAD

**Issue by ghc-mirror** [haskell/haddock#12]
**Original reporter**: _anonymous_

Haddock comments are parsed as separate declarations so we need to insert a ';' when using them with explicit layout:

``` haskell
module NoLayout where {
  -- | the class 'C'
  ;
  g :: Int;
  g = undefined
 }
```

This should be fixed in the GHC parser, hence the 2.3.0 milestone.

**Comment by gbaz** [comment]
still valid.

**Comment by massysett** [comment]
I'm experiencing this bug with GHC 8.8.2 and Haddock 2.23.0.

**Comment by Kleidukos** [comment]
Fixed with 2.26.1
